packages/core/core-flows/src/definition/cart/workflows/confirm-variant-inventory.ts

```diff
@@ -0,0 +1,96 @@
+import { ConfirmVariantInventoryWorkflowInputDTO } from "@medusajs/types"
+import { MedusaError, deepFlatMap, isDefined } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { confirmInventoryStep } from "../steps"
+import { prepareConfirmInventoryInput } from "../utils/prepare-confirm-inventory-input"
+
+export const confirmVariantInventoryWorkflowId = "confirm-item-inventory"
+export const confirmVariantInventoryWorkflow = createWorkflow(
+  confirmVariantInventoryWorkflowId,
+  (input: WorkflowData<ConfirmVariantInventoryWorkflowInputDTO>) => {
+    const confirmInventoryInput = transform({ input }, (data) => {
+      const productVariantInventoryItems = new Map<string, any>()
+      const priceNotFound: string[] = []
+      const stockLocationIds = new Set<string>()
+      const allVariants = new Map<string, any>()
+      let hasSalesChannelStockLocation = false
+      let hasManagedInventory = false
+
+      const salesChannelId = data.input.sales_channel_id
+
+      deepFlatMap(
+        data.input,
+        "variants.inventory_items.inventory.location_levels.stock_locations.sales_channels",
+        ({ variants, inventory_items, stock_locations, sales_channels }) => {
+          if (
+            !hasSalesChannelStockLocation &&
+            sales_channels?.id === salesChannelId
+          ) {
+            hasSalesChannelStockLocation = true
+          }
+
+          if (!isDefined(variants.calculated_price)) {
+            priceNotFound.push(variants.id)
+          }
+
+          stockLocationIds.add(stock_locations.id)
+
+          const inventoryItemId = inventory_items.inventory_item_id
+          if (!productVariantInventoryItems.has(inventoryItemId)) {
+            productVariantInventoryItems.set(inventoryItemId, {
+              variant_id: inventory_items.variant_id,
+              inventory_item_id: inventoryItemId,
+              required_quantity: inventory_items.required_quantity,
+            })
+          }
+
+          if (!allVariants.has(variants.id)) {
+            if (!hasManagedInventory && variants.manage_inventory) {
+              hasManagedInventory = true
+            }
+
+            allVariants.set(variants.id, {
+              id: variants.id,
+              manage_inventory: variants.manage_inventory,
+            })
+          }
+        }
+      )
+
+      if (!hasManagedInventory) {
+        return { items: [] }
+      }
+
+      if (salesChannelId && !hasSalesChannelStockLocation) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Sales channel ${salesChannelId} is not associated with any stock locations.`
+        )
+      }
+
+      if (priceNotFound.length && !data.input.ignore_price_check) {
```

Comment:
question: It feels a bit weird to perform a price check in this workflow, no?

Reply:
yes, I just transposed the full logic that was in place.
I can move that to another place in the next PR.
Is the cart flow the right place to check that though?

Reply:
I think it was added as an extra safety measure to ensure you are not adding a variant without a price. We can remove it for now and assume that variants have prices, but we might add it back in at some point.

Reply:
I can create an extra step to check that. no problem. and remove this check from here

Reply:
Do you want to leave this for a separate PR, or include this now?